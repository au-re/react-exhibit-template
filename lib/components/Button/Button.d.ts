/** A simple Button using styled components */
declare const Button: import("styled-components").StyledComponent<"button", any, {}, never>;
export default Button;
