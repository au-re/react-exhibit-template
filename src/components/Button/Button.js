import styled from "styled-components";

const Button = styled.button`
  text-align: center;
  line-height: 2.5rem;
  font-size: 1rem;
  font-family: sans-serif;
  font-weight: 700;
  padding: 0 12px;
  transition: all 0.2s;
  cursor: pointer;
  border: none;
  border-radius: 2px;
  color: white;
  background: palevioletred;

  :hover {
    background: blueviolet;
  }
`;

export default Button;
