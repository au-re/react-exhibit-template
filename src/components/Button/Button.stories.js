import { storiesOf } from "@storybook/react";
import React, { Fragment } from "react";
import { withDocs } from "storybook-readme";

import Button from "./Button";
import readme from "./README.md";

storiesOf("Button", module)
  .addDecorator(withDocs(readme))
  .add("a simple button", () =>
    <Fragment>
      <Button>default</Button>
    </Fragment>);
