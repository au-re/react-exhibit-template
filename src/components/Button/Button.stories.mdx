import Button from "./Button";

<Meta title="Components|Button" />

Button
===

A simple Button to showcase the library.

```jsx
  <Button>default</Button>
```

<Button>default</Button>