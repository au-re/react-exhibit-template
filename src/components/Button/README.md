Button
===

A simple Button to showcase the library.

```jsx
  <Button>default</Button>
```

<!-- STORY -->