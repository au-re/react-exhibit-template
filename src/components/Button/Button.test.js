import "jest-styled-components";

import React from "react";
import renderer from "react-test-renderer";

import Button from "./Button";

describe("Button", () => {

  it("renders without props", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
