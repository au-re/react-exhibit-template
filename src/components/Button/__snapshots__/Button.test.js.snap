// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without props 1`] = `
.c0 {
  text-align: center;
  line-height: 2.5rem;
  font-size: 1rem;
  font-family: sans-serif;
  font-weight: 700;
  padding: 0 12px;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  border: none;
  border-radius: 2px;
  color: white;
  background: palevioletred;
}

.c0:hover {
  background: blueviolet;
}

<button
  className="c0"
/>
`;
